bitwarden_license/bit-web/src/app/secrets-manager/service-accounts/dialog/service-account-dialog.component.ts

```diff
@@ -39,8 +49,28 @@ export class ServiceAccountDialogComponent implements OnInit {
   ) {}
 
   async ngOnInit() {
-    this.projects = await this.projectService.getProjects(this.data.organizationId);
-    this.secrets = await this.secretService.getSecrets(this.data.organizationId);
+    if (this.data.operation == OperationType.Add) {
+      this.projects = await this.projectService.getProjects(this.data.organizationId);
+      this.secrets = await this.secretService.getSecrets(this.data.organizationId);
+    } else {
+      this.showProjectsSecretsInfo = false;
+      this.loadData();
+    }
+  }
+
+  async loadData() {
+    this.loading = true;
+    const serviceAccount: ServiceAccountView =
+      await await this.serviceAccountService.getByServiceAccountId(
+        this.data.serviceAccountId,
+        this.data.organizationId
+      );
+    this.formGroup.get("name").setValue(serviceAccount.name);
```

Comment:
I personally prefer patchValue in this cases.

```suggestion
    this.formGroup.patchValue({ name: serviceAccount.name});
```